naming/src/main/java/org/jboss/as/naming/management/JndiViewOperation.java

```diff
@@ -148,7 +148,7 @@ private void addEntries(final ModelNode current, final Context context) throws N
                            NamingLogger.ROOT_LOGGER.failedToLookupJndiViewValue(pair.getName(),e);
                        }
                     } else if (!(value instanceof ManagedReferenceFactory)) {
-                       jndiViewValue = String.valueOf(value);
+                       jndiViewValue = (String) value;
```

Comment:
This is not valid: String.valueOf(null) returns a String with the word "null". (String) null returns a null reference.